libpod/container_internal_common.go

```diff
@@ -1940,7 +1940,11 @@ func (c *Container) makeBindMounts() error {
 	if !hasRunContainerenv {
 		// check in the spec mounts
 		for _, m := range c.config.Spec.Mounts {
-			if m.Destination == "/run/.containerenv" || m.Destination == "/run" {
+			switch {
+			case m.Destination == "/run/.containerenv":
+				hasRunContainerenv = true
+				break
+			case m.Destination == "/run" && m.Source != "tmpfs":
```

Comment:
should this check for type tmpfs? The source could be anything when tmpfs is used.

Reply:
yup

Reply:
This was not addressed, please fix it!#